include/openthread/link.h

```diff
@@ -216,6 +217,34 @@ typedef struct otMacCounters
      */
     uint32_t mTxRetry;
 
+    /**
+     * Histogram of number of retries for a single direct packet until success
+     * [0 retry: packet count, 1 retries: packet count, 2 retries : packet count, ...,
+     * n retries : packet count (n == max retry limit)]
+     * The size of the array is OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT.
+     */
+    uint32_t mTxDirectRetrySuccess[OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT];
+
+    /**
+     * The total number of unique MAC transmission packets that meet maximal retry limit for direct packets.
+     *
+     */
+    uint32_t mTxDirectMaxRetryExpiry;
+
+    /**
+     * Histogram of number of retries for a single indirect packet until success
+     * [0 retry: packet count, 1 retries: packet count, 2 retries : packet count, ...,
+     * n retries : packet count (n == max retry limit)]
+     * The size of the array is OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_INDIRECT.
+     */
+    uint32_t mTxIndirectRetrySuccess[OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_INDIRECT];
```

Comment:
How about if we provide APIs to get these counter info instead of array definition in `otMacCounter` to allow the size be flexible.

Something like this:
```
const uint32_t *otLinkGetTxDirectRetrySuccessCounter(uint8_t &aNumberOfEntries);
```
We get a pointer to array of counter and size of array is ouputeed from `aNumberOfEntries`.